File: java/ankit_kanyal/devnest_problems/RotateList.java
// link- https://devsnest.in/algo-challenges/rotate-list

static Node solve(Node head, int k){
//CODE HERE 

int length=0;
Node cur=head;

while(cur!=null){
    cur=cur.next;
    length++;
}
k=k%length;
if(k==0) return head;
cur=head;
while(length-k>1){
    cur=cur.next;
    length--;
}
Node rotate=cur.next;
cur.next=null;
cur=rotate;
while(cur.next!=null){
    cur=cur.next;
}
cur.next=head;

return rotate;
}

File: Python/chandu/Rotate_List.py
def solve(head, k):
    # CODE HERE
    if not head or not head.next or k==0:
        return head
    
    l = 1
    itr = head
    while itr.next:
        l += 1
        itr = itr.next
    
    itr.next = head
    rotate = int(k%l)
    r = l - rotate

    while r:
        itr = itr.next
        r -= 1
    
    head = itr.next
    itr.next = None

    return head
